File: 103-binary-tree-zigzag-level-order-traversal/103-binary-tree-zigzag-level-order-traversal.java
class Solution {
    public List<List<Integer>> zigzagLevelOrder(TreeNode root) {
        var q = new LinkedList<TreeNode>();
        var levels = new LinkedList<List<Integer>>();
        var lvl = 0;
        if (root != null) q.add(root);
        while (!q.isEmpty()) {
            var sz = q.size();
            var level = new LinkedList<Integer>();
            for (int i = 0; i < sz; i++) {
                var x = q.poll();
                if (lvl % 2 == 0) level.add(x.val);
                else level.addFirst(x.val);
                if (x.left != null) q.add(x.left);
                if (x.right != null) q.add(x.right);
            }
            levels.add(level);
            lvl++;
        }
        return levels;
    }
}

File: 103-binary-tree-zigzag-level-order-traversal/103-binary-tree-zigzag-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def zigzagLevelOrder(self, root):
        levels, level = [], [root] if root else []
        n = 0
        while level:
            levels.append([n.val for n in level[::(-1, 1)[n % 2 == 0]]])
            level = [c for n in level for c in [n.left, n.right] if c]
            n += 1
        return levels